shell/platform/common/public/flutter_messenger.h

```diff
@@ -87,6 +88,47 @@ FLUTTER_EXPORT void FlutterDesktopMessengerSetCallback(
     FlutterDesktopMessageCallback callback,
     void* user_data);
 
+// Increments the reference count for the |messenger|.
+//
+// See also: |FlutterDesktopMessengerRelease|
+FLUTTER_EXPORT FlutterDesktopMessengerRef
+FlutterDesktopMessengerAddRef(FlutterDesktopMessengerRef messenger);
+
+// Decrements the reference count for the |messenger|.
+//
+// The reference count starts at zero so it is valid to call
+// FlutterDesktopMessengerRelease without having called
+// FlutterDesktopMessengerAddRef.
+//
+// See also: |FlutterDesktopMessengerAddRef|
+FLUTTER_EXPORT void FlutterDesktopMessengerRelease(
```

Comment:
What thread(s) can adding and releasing be done on?

Reply:
Thread-safe, added the note to the docstring